reconcile/cna/assets/null.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Any, Mapping, Optional
+
+from reconcile.cna.assets.asset import Asset, AssetError, AssetStatus, AssetType
+from reconcile.gql_definitions.cna.queries.cna_resources import CNANullAssetV1
+
+
+@dataclass
+class NullAsset(Asset):
+    addr_block: Optional[str]
+
+    def api_payload(self) -> dict[str, Any]:
+        return {
+            "asset_type": "null",
+            "name": self.name,
+            "parameters": {
+                "addr_block": self.addr_block,
+            },
+        }
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
would it be an option to use frozen dataclasses to get eq, ne and hash for free? from what i see, you don't update fields right now but create new instances instead, e.g. in `update_from`. so should be ok?

Reply:
The `__eq__` method gets overridden to ensure only `name` and `addr_block` are compared. We do not care about comparing ids, herfs, ... . This comes in handy when comparing equality of states: desired state objects (taken from app-interface) do not contain any ids, but current states objects, taken from CNA API do contain ids, still the states can be equal, regardless of their ids. 
The `__hash__` method is a bad side-effect of this implementation. It is required, because we use a custom `__eq__` methods. 

However, I actually do not like this implementation a lot and I am very happy about any suggestion here. But Im not sure how frozen dataclass can solve this.

Reply:
oh i understand. so you would like to compare only those fields that come from app-interface?

Reply:
have you tried `dataclass.Field.compare` and `dataclass.Field.hash` ?

Reply:
> have you tried dataclass.Field.compare and dataclass.Field.hash ?

Interesting. Looking into it

Reply:
Yeah looks much better with `dataclass.Field.compare` and `dataclass.Field.hash`. Thank you! :bow: